### PR TITLE
Feat : Edit project

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: src/components/New/Cards/HorizontalListItemCard.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply src/components/New/Cards/HorizontalListItemCard.tsx]: fixed

File: src/components/New/Cards/HorizontalListItemCard.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+interface HorizontalListItemCardProps {
+  imageUrl: string;
+  title: string;
+  linkUrl: string;
+}
+
+const HorizontalListItemCard: React.FC<HorizontalListItemCardProps> = ({ imageUrl = "", title = "", linkUrl = "" }) => {
+  return (
+    <Card className="shadow-medium" key={title}>
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply src/components/New/Cards/HorizontalListItemCard.tsx]: fixed

File: src/pages/ProjectDetails/Sidebar/ProjectOption.tsx [hunk 3]
```diff
@@ -3,34 +3,35 @@ import ShootingStar from "src/assets/icons/ShootingStar";
...
-export default function ProjectOption({ project, isSelected }: Props) {
+export default function ProjectOption({ project, isSelected, isInvited }: Props) {
   const { T } = useIntl();
 
   return (
     <Listbox.Option
       value={project}
       className={`border-neutral-600 p-4 duration-300 last:rounded-b-2xl hover:cursor-pointer hover:bg-white/10 ${
-        project.withInvitation && "bg-orange-400/10  hover:bg-amber-700/30"
+        isInvited && "bg-orange-400/10  hover:bg-amber-700/30"
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Sidebar/ProjectOption.tsx:21]:
```suggestion
        isInvited && "bg-orange-400/10 hover:bg-amber-700/30"
```

File: src/pages/ProjectDetails/Sidebar/ProjectOption.tsx [hunk 4]
```diff
@@ -3,34 +3,35 @@ import ShootingStar from "src/assets/icons/ShootingStar";
...
 }
 
-export default function ProjectOption({ project, isSelected }: Props) {
+export default function ProjectOption({ project, isSelected, isInvited }: Props) {
   const { T } = useIntl();
 
   return (
     <Listbox.Option
       value={project}
       className={`border-neutral-600 p-4 duration-300 last:rounded-b-2xl hover:cursor-pointer hover:bg-white/10 ${
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Sidebar/ProjectOption.tsx:20]: Could use `cn` util

File: src/pages/ProjectDetails/Sidebar/ProjectOption.tsx [hunk 5]
```diff
@@ -3,34 +3,35 @@ import ShootingStar from "src/assets/icons/ShootingStar";
...
-        project.withInvitation && "bg-orange-400/10  hover:bg-amber-700/30"
+        isInvited && "bg-orange-400/10  hover:bg-amber-700/30"
       } `}
     >
       <div className="flex flex-col gap-5">
         <div className="flex flex-row items-center gap-5">
-          <RoundedImage src={project.logoUrl} alt="Project Logo" />
+          <RoundedImage src={project.logoUrl || onlyDustLogo} alt="Project Logo" />
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Sidebar/ProjectOption.tsx:26]: This logic is used often, maybe add a `useFallback` prop for this

File: src/pages/ProjectDetails/Sidebar/ProjectOption.tsx [hunk 6]
```diff
@@ -40,7 +41,7 @@ export default function ProjectOption({ project, isSelected }: Props) {
             )}
           </>
         </div>
-        {project.withInvitation && (
+        {isInvited && (
           <div className="w-full rounded-xl bg-neutral-100 p-2 text-center text-sm text-black">View invite</div>
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Sidebar/ProjectOption.tsx:45]: Translation

File: src/pages/ProjectDetails/Sidebar/View.tsx [hunk 7]
```diff
@@ -10,33 +10,39 @@ import { cn } from "src/utils/cn";
...
+  availableTabs,
+  onLinkClick,
+  pendingProjects,
+  projects,
+}: Props) {
   const { T } = useIntl();
   const navigate = useNavigate();
   const isXl = useMediaQuery(`(min-width: ${viewportConfig.breakpoints.xl}px)`);
 
+  console.log("props", {
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Sidebar/View.tsx:37]: Remove

File: src/pages/ProjectDetails/Contributors/index.tsx [hunk 8]
```diff
@@ -74,25 +77,28 @@ export default function Contributors() {
         <div className="flex flex-row items-center justify-between gap-2">
           {T("project.details.contributors.title")}
           {isProjectLeader && !isFetching && (
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Contributors/index.tsx:79]:
```suggestion
          {isProjectLeader && (
```

File: src/types.ts [hunk 9]
```diff
@@ -319,3 +321,16 @@ export type Sorting = {
   field: string | undefined;
   isAscending: boolean | undefined;
 };
+
+type repos = components["schemas"]["ShortGithubRepoResponse"] & {
```
Comment (haydencleary) [inline src/types.ts:325]:
```suggestion
type Repos = components["schemas"]["ShortGithubRepoResponse"] & {
```

File: src/types.ts [hunk 10]
```diff
@@ -319,3 +321,16 @@ export type Sorting = {
...
+  selected?: boolean;
+};
+
+type organization = Omit<components["schemas"]["GithubOrganizationResponse"], "repos"> & {
```
Comment (haydencleary) [inline src/types.ts:329]:
```suggestion
type Organization = Omit<components["schemas"]["GithubOrganizationResponse"], "repos"> & {
```

File: src/components/RewardTable/RewardTable.tsx [hunk 11]
```diff
@@ -43,7 +44,7 @@ export default function RewardTable({ rewards, options, projectId }: RewardTable
     try {
       // refetch PaymentRequests to hide MyRewards
       client.refetchQueries({ include: ["GetPaymentRequestIds"] });
-      queryClient.invalidateQueries({ queryKey: ["GetUser"] });
+      queryClient.invalidateQueries({ queryKey: MeApi.tags.all });
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply src/components/RewardTable/RewardTable.tsx:48]: We need to refactor the Delete reward mutation and invalidate the query directly in it

File: src/pages/ProjectDetails/ProjectEdition/utils/transformInstallationToOrganization.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import { components } from "src/__generated/api";
+
+export default function transformOrganization(
+  input: components["schemas"]["InstallationResponse"] | undefined
+): components["schemas"]["ProjectGithubOrganizationResponse"] | undefined {
+  if (!input) {
```
[the remark this answers is not in the document]
Comment (pixelfact) [reply src/pages/ProjectDetails/ProjectEdition/utils/transformInstallationToOrganization.ts]: fixed

**Pull request merged**